src/Structures/Queue.ts

```diff
@@ -175,14 +176,14 @@ class Queue<T = unknown> {
             this.playing = true;
             if (!this._filtersUpdate && resource?.metadata) this.player.emit("trackStart", this, resource?.metadata ?? this.current);
             this._filtersUpdate = false;
+            this._trackSkipped = false;
         });
 
         this.connection.on("finish", async (resource) => {
             if (this.#watchDestroyed(false)) return;
             this.playing = false;
             if (this._filtersUpdate) return;
             this._streamTime = 0;
-            if (resource && resource.metadata) this.previousTracks.push(resource.metadata);
```

Comment:
why did you remove this?

Reply:
On line 603 (in my branch) in the play function:
```javascript
        if (!options.filtersUpdate) {
            this.previousTracks = this.previousTracks.filter((x) => x.id !== track.id);
            this.previousTracks.push(track);
        }
```
There was a duplication of the track being pushed onto previousTracks. I removed the line to rectify this issue. It's possible to have removed the one on line 603 and kept the one I removed in my branch, but some existing functionality would need to be updated to account for the current song not existing in either previousTracks or queue until it finished.